[BTC->ERC20] Both parties to BothFunded state 💰️

LLFourn opened the pull request:
As described

fixes #435 
Note that issue was only up to BetaDeployed but I was on a roll! XD

LLFourn committed:
[BTC->ERC20] Both parties to BothFunded state 💰️

D4nte reviewed (changes requested):
Few changes regarding how we transformed the data returned by actions.
I believe we should not need to transform it.

I would go as far as saying that maybe the `gas_limit` and the `value` should both me decimal (or both hex).

Commit:
Reorder actions in from_str

Commit:
React to PR comments

- Modify `send_eth_transaction_to` to accept a string `value`
  parameter as opposed to a hex to then transform it inside the
  function to BN and hex using `web3.utils.numberToHex`.
- Add `gas_limit` parameter to `deploy_eth_contract`.
- Increase `Erc20Htlc` gas limit (the previous estimate was ever
  so slightly below the required amount)

D4nte reviewed (approved):
Well done both!
